### PR TITLE
Test finite mps

### PR DESCRIPTION
**Pull request by MichaelMarien**
* added some tests for finite mps (now at 100% cov)
* removed check_canconical implementation in finite_mps as it is identical to parent class

### CODE REVIEW

File: tensornetwork/matrixproductstates/finite_mps.py [hunk 1]
```diff
@@ -132,22 +132,6 @@ def canonicalize(self, normalize: Optional[bool] = True) -> np.number:
     self.position(len(self.nodes) - 1, normalize=False)
     return self.position(pos, normalize=normalize)
 
-  def check_canonical(self) -> Tensor:
```
Comment (Thenerdstation) [inline tensornetwork/matrixproductstates/finite_mps.py:135]: Why is this deleted?

Comment (MichaelMarien) [reply tensornetwork/matrixproductstates/finite_mps.py:135]: Perhaps a bit over-aggressive, but it's the exact same implementation as the method in the parent class (BaseMPS). Don't mind putting it back though.

Comment (Thenerdstation) [reply tensornetwork/matrixproductstates/finite_mps.py:135]: That's what I assumed, just making sure. Let keep it deleted.
